### PR TITLE
Upgrade instant v3

### PR DESCRIPTION
**Pull request by dave4506**
## Description

Entailed in this PR:
- Removal of asset-buyer from the monorepo
- Refactor of instant to use asset-swapper over asset-buyer (thereby upgrading it to v3)
- Temporarily disabled affiliate fees in instant until a better long-term solution is developed for the fees related with the 0x protocol
- Minor ergonomic improvements to asset-swapper after 'dogfooding' asset-swapper with the instant refactor. 
  - Addition of `gasPrice` to the `SwapQuote` object
  - Addition of a getWethAssetData... convenience function on `SwapQuoter`
  - Minor changes to how options are shaped in `SwapQuoteConsumer`

## Testing instructions

While these refactors have past the unit tests, we will need to test comprehensively the UI of instant with the new protocol fee.
- Test takerFee in both makerAsset and takerAsset
- Test protocolFee with gasPrice
- Try with NFT
- Test for corner cases when gasPrice change or the user changes it in Metamask? 

## Checklist:

-   [ ] Comprehensive testing of orders in instant (specially how the fees are displayed to the user)
-   [ ] Mesh support? (Since swapper supports it)
-   [x] Possibly need to wait on the v3 redeploy with the change to rounding in marketBuy to test precise amounts in orders.

### TODO
- [ ] NFT
- [x] Affiliate fees
- [ ] order fees
- [x] Fix in-browser DevUtils. Strict mode and usage of `this` in merkle-patricia-tree library https://github.com/ethereumjs/merkle-patricia-tree/pull/42

### CODE REVIEW

File: packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts [hunk 1]
```diff
@@ -78,21 +87,26 @@ const mapDispatchToProps = (
...
+        txHash: string,
+        startTimeUnix: number,
+        expectedEndTimeUnix: number,
+    ) => {
+        dispatch(actions.setSwapOrderStateProcessing(txHash, startTimeUnix, expectedEndTimeUnix));
     },
-    onBuySuccess: (buyQuote: BuyQuote, txHash: string) => dispatch(actions.setBuyOrderStateSuccess(txHash)),
-    onBuyFailure: (buyQuote: BuyQuote, txHash: string) => dispatch(actions.setBuyOrderStateFailure(txHash)),
+    onBuySuccess: (swapQuote: MarketBuySwapQuote, txHash: string) => dispatch(actions.setSwapOrderStateSuccess(txHash)),
+    onBuyFailure: (swapQuote: MarketBuySwapQuote, txHash: string) => dispatch(actions.setSwapOrderStateFailure(txHash)),
```
Comment (fragosti) [inline packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts]: A good follow-up here would be to show an error specific to the protocol fee failure edge cases.

File: packages/instant/src/index.umd.ts [hunk 2]
```diff
@@ -172,30 +173,31 @@ export const hasMetaDataForAssetData = (assetData: string): boolean => {
...
 
-    const assetBuyerOptions = { networkId };
+    const swapQuoterOptions = { chainId };
 
-    const assetBuyer = _.isString(orderSource)
-        ? AssetBuyer.getAssetBuyerForStandardRelayerAPIUrl(bestProvider, orderSource, assetBuyerOptions)
-        : AssetBuyer.getAssetBuyerForProvidedOrders(bestProvider, orderSource, assetBuyerOptions);
+    const swapQuoter = _.isString(orderSource)
+        ? SwapQuoter.getSwapQuoterForStandardRelayerAPIUrl(bestProvider, orderSource, swapQuoterOptions)
+        : SwapQuoter.getSwapQuoterForProvidedOrders(bestProvider, orderSource, swapQuoterOptions);
```
Comment (fragosti) [inline packages/instant/src/index.umd.ts]: Should use new factory util here.

File: packages/instant/webpack.config.js [hunk 3]
```diff
@@ -150,7 +150,16 @@ const generateConfig = (dischargeTarget, heapConfigOptions, rollbarConfigOptions
...
+                    use: [
+                        {
+                            loader: 'react-svg-loader',
+                            options: {
+                                svgo: {
+                                    plugins: [{ removeViewBox: false }],
+                                },
+                            },
+                        },
+                    ],
```
Comment (fragosti) [inline packages/instant/webpack.config.js]: 👍🏼

File: packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts (hunk 1 above)
Comment (dave4506) [reply packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts]: I'll tentatively add a TODO for this, one large feedback at EthWaterloo is we need richer errors.

File: packages/asset-swapper/src/utils/protocol_fee_utils.ts [hunk 4]
```diff
@@ -1,16 +1,31 @@
...
+            const protocolFeeMultiplier = await this._exchangeContract.protocolFeeMultiplier.callAsync();
+            return protocolFeeMultiplier;
+        }
+    }
+
+    // tslint:disable-next-line: prefer-function-over-method
+    public async getGasPriceEstimationOrThrowAsync(): Promise<BigNumber> {
         try {
             const res = await fetch(`${constants.ETH_GAS_STATION_API_BASE_URL}/json/ethgasAPI.json`);
             const gasInfo = await res.json();
```
[the remark this answers is not in the document]
Comment (dave4506) [reply packages/asset-swapper/src/utils/protocol_fee_utils.ts]: what is the reasoning to that?

**Comment by dave4506** [comment]
@dekz Thanks for making the changes!
